### ISSUE TITLE
Linting ignore for Class/forName

### ISSUE THREAD

**Issue by dotemacs** [clj-kondo/clj-kondo#2316]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**

clj-kondo v2024.03.13

**platform**

macOS & GitHub Action CI runners, which are Linux based

**editor**

Not applicable

**problem**

The code in question:

https://github.com/dotemacs/pdfboxing/blob/8cef6ac9e47d7526e8ce18addeef2273bb392445/src/pdfboxing/common.clj#L41-L43

Returns:
 
```shell
$ clj-kondo --lint src
src/pdfboxing/common.clj:43:18: error: Function name must be simple symbol but got: "[B"
linting took 58ms, errors: 1, warnings: 0
```

**repro**

```
git clone https://github.com/dotemacs/pdfboxing
cd pdfboxing
clj-kondo --lint src
```

To be specific, the commit that the master is on: https://github.com/dotemacs/pdfboxing/commit/8cef6ac9e47d7526e8ce18addeef2273bb392445

**config**

No config.edn: 

```
$ ls -l ~/.clj-kondo/
total 0
```

**expected behavior**

With this:

```clojure
(extend-protocol PDFDocument
  #_:clj-kondo/ignore
  (Class/forName "[B") ;; byte-array
```

I expected the error to be silenced.
